code/game/objects/items/plushes.dm

```diff
@@ -542,3 +542,133 @@
+
+/obj/item/toy/plush/goatplushie/angry
+	var/mob/living/carbon/target
+	throwforce = 6
+	var/cooldown = 0
+	var/cooldown_modifier = 20
+
+/obj/item/toy/plush/goatplushie/angry/Initialize()
+	. = ..()
+	START_PROCESSING(SSprocessing, src)
+
+/obj/item/toy/plush/goatplushie/angry/process()
+	if (prob(25) && !target)
+		var/list/targets_to_pick_from = list()
+		for(var/mob/living/carbon/C in view(7, src))
+			if(considered_alive(C.mind))
+				targets_to_pick_from += C
+		if (!targets_to_pick_from.len)
+			return
+		target = pick(targets_to_pick_from)
+		visible_message("<span class='notice'>[src] stares at [target].</span>")
+	if (world.time > cooldown && target)
+		ram()
+
+/obj/item/toy/plush/goatplushie/angry/proc/ram()
+	if(prob((obj_flags & EMAGGED) ? 98:90) && isturf(loc) && considered_alive(target.mind))
+		throw_at(target, 10, 10)
+		visible_message("<span class='danger'>[src] rams [target]!</span>")
+		cooldown = world.time + cooldown_modifier
+	target = null
+	visible_message("<span class='notice'>[src] looks disinterested.</span>")
+
+/obj/item/toy/plush/goatplushie/angry/emag_act(mob/user)
+	if (obj_flags&EMAGGED)
+		visible_message("<span class='notice'>[src] already looks angry enough, you shouldn't anger it more.</span>")
+		return
+	cooldown_modifier = 5
+	throwforce = 20
+	obj_flags |= EMAGGED
+	visible_message("<span class='danger'>[src] stares at [user] angrily before going docile.</span>")
+
+/obj/item/toy/plush/goatplushie/angry/Destroy()
+	STOP_PROCESSING(SSprocessing, src)
+	return ..()
+
+/obj/item/toy/plush/goatplushie
+	squeak_override = list('sound/items/goatsound.ogg'=1)
+
+/obj/item/toy/plush/goatplushie/angry/realgoat
+	name = "goat plushie"
+	icon_state = "realgoat"
+
+/obj/item/toy/plush/realgoat
+	name = "goat plushie"
+	desc = "Despite its cuddly appearance and plush nature, it will beat you up all the same, or atleast it would if it wasnt a normal plushie."
+	icon_state = "realgoat"
+	squeak_override = list('sound/items/goatsound.ogg'=1)
+
+/obj/item/toy/plush/goatplushie/angry/kinggoat
+	name = "King Goat Plushie"
+	desc = "A plushie depicting the king of all goats."
+	icon_state = "kinggoat"
+	cooldown_modifier = 3
+	throwforce = 25
+	force = 25
+	attack_verb = list("chomped")
+
+/obj/item/toy/plush/goatplushie/angry/ascendedkinggoat
+	name = "Ascended King Goat Plushie"
+	desc = "A plushie depicting the god of all goats."
+	icon_state = "ascendedkinggoat"
+	cooldown_modifier = 1
+	throwforce = 30
+	force = 30
+	attack_verb = list("chomped")
+
+/obj/item/toy/plush/goatplushie/angry/ascendedkinggoat/attackby(obj/item/I,mob/living/user,params)
+	if(I.get_sharpness())
+		user.visible_message("<span class='notice'>[user] attempts to stab [src]!</span>", "<span class='suicide'>The knife bounces off [src]'s back before breaking into millions of pieces... [src] glares at [user]!</span>") // You fucked up now son
```

Comment:
The item is not necessarily a knife.

Reply:
By the way, if you want to use my "plushie says no u" code that I wrote for the ratvar and nar'sie plushies (instead of doing this), you can just set the "divine" variable of the plushie to "TRUE".